pocketlint/formatcheck.py

```diff
@@ -460,6 +460,12 @@ def check_regex_line(self, line_no, line):
                     icon='info',
                     )
 
+def check_semantic_newline(self):
+        """Check for a sentence with no new line."""
+        if self.text.find('. ') and self.text[-1] != '\n':
```

Comment:
self.text is the whole file... as commented below you should pass the line_no and line here and do the search only inside that line.

searching only for `line.find('. ')` should be enough... no need for `line[-1]`

we might also want to search for question mark and exclamation mark.

once this is done, do a full check on our full documentation to see for any false positives